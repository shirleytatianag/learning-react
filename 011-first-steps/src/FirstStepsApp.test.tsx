import {afterEach, describe, expect, test, vi} from "vitest";
import {render, screen} from "@testing-library/react";
import {FirstStepsApp} from "./FisrtStepsApp.tsx";

const mockItemCounter = vi.fn(({name, quantity}: { name: string; quantity: number }) => {
  return (
    <div data-testid="item-counter-mock">
      <span>{name}</span>
      <span>{quantity}</span>
    </div>
  );
})

vi.mock('./shopping-cart/ItemCounter.tsx', () => ({
  ItemCounter: (props: { name: string; quantity: number }) => mockItemCounter(props)
}))

// vi.mock('./shopping-cart/ItemCounter.tsx', () => ({
//   ItemCounter: ({name, quantity}: { name: string; quantity: number }) => {
//     return (
//       <div data-testid="item-counter-mock">
//         <span>{name}</span>
//         <span>{quantity}</span>
//       </div>
//     );
//   }
// }))

describe("FirstStepsApp", () => {
    afterEach(() => {
      vi.clearAllMocks()
    })

    test("should match snapshot", () => {
      const {container} = render(<FirstStepsApp/>);
      expect(container).toMatchSnapshot();
    })

    test("should render the correct number of ItemCounter components", () => {
      render(<FirstStepsApp/>);
      const itemCounters = screen.getAllByTestId("item-counter-mock");
      expect(itemCounters.length).toBe(3);
    })

    test("should render the correct number of ItemCounter components", () => {

      render(<FirstStepsApp/>);
      expect(mockItemCounter).toHaveBeenCalledTimes(3);
      expect(mockItemCounter).toHaveBeenCalledWith({name: 'Nintendo Switch 2', quantity: 1});
      expect(mockItemCounter).toHaveBeenCalledWith({name: 'Pro Controller', quantity: 2});
      expect(mockItemCounter).toHaveBeenCalledWith({name: 'Super Smash', quantity: 5});
      screen.debug()
    })
  }
);