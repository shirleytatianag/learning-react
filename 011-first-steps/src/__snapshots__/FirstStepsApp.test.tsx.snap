// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FirstStepsApp > should match snapshot 1`] = `
<div>
  <h1>
    Carrito de compras
  </h1>
  <div
    data-testid="item-counter-mock"
  >
    <span>
      Nintendo Switch 2
    </span>
    <span>
      1
    </span>
  </div>
  <div
    data-testid="item-counter-mock"
  >
    <span>
      Pro Controller
    </span>
    <span>
      2
    </span>
  </div>
  <div
    data-testid="item-counter-mock"
  >
    <span>
      Super Smash
    </span>
    <span>
      5
    </span>
  </div>
</div>
`;
